### ISSUE TITLE
Breaking change in .NET 8 means instructions to run docker container are wrong

### ISSUE THREAD

**Issue by jonathanpickard3sq** [dotnet/core#8958]
Problem encountered on https://dotnet.microsoft.com/en-us/learn/aspnet/microservice-tutorial/run-docker
Operating System: windows

The instructions in the tutorial are not correct for somebody who has created the microservice using .NET 8.  [Breaking change - default port was changed from 80 to 8080 ](https://learn.microsoft.com/en-us/dotnet/core/compatibility/containers/8.0/aspnet-port)

There are potentially two fixes for this: either change the command line to run the container:

`docker run -it --rm -p 3000:8080 --name mymicroservicecontainer mymicroservice`

Or change the docker file to enforce the internal container port to port 80 (and leave the command-line as-is):

```
FROM mcr.microsoft.com/dotnet/sdk:8.0 AS build
WORKDIR /src
COPY MyMicroservice.csproj .
RUN dotnet restore
COPY . .
RUN dotnet publish -c release -o /app

FROM mcr.microsoft.com/dotnet/aspnet:8.0
WORKDIR /app
COPY --from=build /app .
ENTRYPOINT ["dotnet", "MyMicroservice.dll"]

ENV ASPNETCORE_HTTP_PORTS=80

```

**Comment by rzikm** [comment]
@mairaw We need to update the website, can you please have a look?

**Comment by mairaw** [comment]
Yes, thank you for assigning this to me. It was on my to-do list to review this with @bradygaster and @baronfel.

**Comment by baronfel** [comment]
There are a few options, ranking them in my personal order of preference:

### Let Docker auto-assign ports 

Run the docker command with `-P` instead of an explicit port, then use a command like `docker container list` to see what port was bound by Docker and tell the user to go there:
```terminal
$docker run -it -d -P registry:2
cdaf46d3dae95c16d84e1b832b0a2b19d5fbf6810cc4aa6e91b8c9d828b84dbd
$docker container list
CONTAINER ID   IMAGE        COMMAND                  CREATED         STATUS        PORTS                     NAMES
cdaf46d3dae9   registry:2   "/entrypoint.sh /etc…"   2 seconds ago   Up 1 second   0.0.0.0:32768->5000/tcp   dreamy_lamport
```
in this case you'd tell the user to go to localhost:32768, for example. This has the advantage of being completely resilient to any changes in the base images, but it introduces more concepts up-front.

### Change the `run` command to bind to the new 8080 port

This is what @jonathanpickard3sq describes above. Change `-p 3000:80` to `-p 3000:8080`. This is the smallest possible change but it does leave you vulnerable to any other changes from the base image. Now that we're capable of running rootless though I'm not sure we'd have many of these kinds of changes.

### Change the actual generation of the container to force the port back to 80

This is my least favorite - we changed the port for good reason and making this change defeats the 'rootless' story.


As a separate question, do we want to change this tutorial to make the container using the SDK? That would remove the entire section about creating Dockerfiles and replace the docker build command with `dotnet publish -t:PublishContainer` to get the same final result.
